From f657c842ad29e0fd52e01f63d9399c50bc391529 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 10 Dec 2020 11:49:39 +0000
Subject: [PATCH] GCC: remove use of (std::stringstream() << x).str()

Operator << of std::stringstream returns a reference to std::ostream
which has no member function str().

Missed while fixing
https://chromium-review.googlesource.com/c/chromium/src/+/2576877
---
 chrome/browser/web_applications/external_web_app_manager.cc | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/chrome/browser/web_applications/external_web_app_manager.cc b/chrome/browser/web_applications/external_web_app_manager.cc
index a33e4bc..d84d08f 100644
--- a/chrome/browser/web_applications/external_web_app_manager.cc
+++ b/chrome/browser/web_applications/external_web_app_manager.cc
@@ -22,6 +22,7 @@
 #include "base/no_destructor.h"
 #include "base/path_service.h"
 #include "base/stl_util.h"
+#include "base/strings/strcat.h"
 #include "base/task/post_task.h"
 #include "base/task/thread_pool.h"
 #include "base/threading/scoped_blocking_call.h"
@@ -100,9 +101,8 @@ LoadedConfigs LoadConfigsBlocking(const base::FilePath& config_dir) {
     std::unique_ptr<base::Value> app_config =
         deserializer.Deserialize(nullptr, &error_msg);
     if (!app_config) {
-      result.errors.push_back(
-          (std::stringstream() << file << " was not valid JSON: " << error_msg)
-              .str());
+      result.errors.push_back(base::StrCat({file.AsUTF8Unsafe(),
+                                           " was not valid JSON: ", error_msg}));
       VLOG(1) << result.errors.back();
       continue;
     }
-- 
2.26.2

